Mathlib/Order/Hom/Basic.lean

```diff
@@ -1307,3 +1307,19 @@ lemma denselyOrdered_iff_of_orderIsoClass {X Y F : Type*} [Preorder X] [Preorder
     exact ⟨EquivLike.inv f c, by simpa using hc⟩
 
 end DenselyOrdered
+
+/-- `Fin (n + 1)` is order equivalent to `Fin n` with a bottom element. -/
+def WithBot.orderIsoFin (n : ℕ) : WithBot (Fin n) ≃o Fin (n + 1) where
```

Comment:
We already have the `Equiv` version in `finSuccEquiv`. Why not reuse it? Imports? Something else?